curve25519-dalek/src/lizard/lizard_ristretto.rs

```diff
@@ -178,33 +180,43 @@ impl RistrettoPoint {
         ]
     }
 
-    /// Interprets the given bytestring as a positive field element and computes the Ristretto
-    /// Elligator map. Note this clears the bottom bit and top two bits of `bytes`.
+    /// Clears bits in `bytes` to make it a positive, reduced field element, then performs
+    /// [`RistrettoPoint::map_to_curve`]). Specifically, clears the the bottom bit (bottom bit of
+    /// `bytes[0]`) and second-to-top bit (second-to-top bit of `bytes[31]`). The first is to ensure
+    /// we have a positive field element and the second is to ensure we are below the modulus
+    /// (map-to-curve clears the topmost bit for us).
+    pub fn map_pos_felem_to_curve(mut bytes: [u8; 32]) -> RistrettoPoint {
+        bytes[0] &= 0b11111110;
+        bytes[31] &= 0b10111111;
+
+        RistrettoPoint::map_to_curve(bytes)
+    }
+
+    /// Interprets the given bytestring as a field element and computes the Ristretto Elligator map.
+    /// This is the MAP function in
+    ///     [RFC 9496](https://www.rfc-editor.org/rfc/rfc9496.html#section-4.3.4-4).
+    /// Note this clears the top bit (`bytes[31] & 0x80`).
     ///
     /// # Warning
     ///
     /// This function does not produce cryptographically random-looking Ristretto points. Use
     /// [`Self::hash_from_bytes`] for that. DO NOT USE THIS FUNCTION unless you really know what
     /// you're doing.
     pub fn map_to_curve(mut bytes: [u8; 32]) -> RistrettoPoint {
-        // We only have a meaningful inverse if we give Elligator a point in its domain, ie a
-        // positive (meaning low bit 0) field element. Mask off the top two bits to ensure it's less
-        // than the modulus, and the bottom bit for evenness.
-        bytes[0] &= 0b11111110;
-        bytes[31] &= 0b00111111;
+        bytes[31] &= 0b01111111;
```

Comment:
This now ought to be redundant with all current implementations of `FieldElement::from_bytes`, but I can see that it might be safer not to depend on that.